src/extractor/geojson_debug_policies.cpp

```diff
@@ -65,5 +67,28 @@ operator()(const NodeID intersection_node,
     return features;
 }
 
+IntersectionPrinter::IntersectionPrinter(
+    const util::NodeBasedDynamicGraph &node_based_graph,
+    const std::vector<extractor::QueryNode> &node_coordinates,
+    const extractor::guidance::CoordinateExtractor &coordinate_extractor)
+    : IntersectionShapePrinter(node_based_graph, node_coordinates, coordinate_extractor)
+{
+}
+
+util::json::Array IntersectionPrinter::
+operator()(const NodeID intersection_node,
+           const extractor::guidance::Intersection &intersection,
+           const boost::optional<util::json::Object> &node_style,
+           const boost::optional<util::json::Object> &way_style) const
+{
+    guidance::IntersectionShape intersection_shape;
+    intersection_shape.resize(intersection.size());
+    std::transform(intersection.begin(),
+                   intersection.end(),
+                   intersection_shape.begin(),
+                   [](const auto &road) { return guidance::IntersectionShapeData(road); });
```

Comment:
Hmmm let's see if I get this right:
- intersection if of type Intersection, which is a vector of ConnectedRoads
- ConnectedRoad is a IntersectionViewData
- IntersectionViewData is a IntersectionShapeData

you upcast through three layers here slicing away the ConnectedRoad's structure leaving you with a IntersectionShapeData (?)

Hmm strongly couples the implementation here with the intersection representation a bit ..